### PR TITLE
Install libspatialindex from the install script

### PR DESCRIPTION
**Pull request by miklos1**
Ready to go.

### CODE REVIEW

File: scripts/firedrake-install [hunk 1]
```diff
@@ -588,6 +588,30 @@ def build_and_install_glpsol():
         os.chdir("..")
 
 
+def build_and_install_libspatialindex():
+    log.info("Installing libspatialindex\n")
+    if os.path.exists("libspatialindex"):
+        libspatialindex_changed = git_update("libspatialindex")
+    else:
+        git_clone("git+https://github.com/miklos1/libspatialindex.git")
```
Comment (miklos1) [inline scripts/firedrake-install]: We will change this to `github.com/firedrakeproject/libspatialindex` if everything goes well.

File: docs/source/download.rst [hunk 2]
```diff
@@ -165,7 +164,6 @@ On Ubuntu, the following apt packages are required:
 - libblas-dev
 - liblapack-dev
 - libopenmpi-dev
-- libspatialindex-dev
```
Comment (wence-) [inline docs/source/download.rst]: If you're installing from individual components (and not using the install script) then you do still need spatialindex somehow I think.

Comment (miklos1) [inline docs/source/download.rst]: Right, I need to add that somewhere. The system provided package won't be suitable in the near future, as we will switch to the C API.

Comment (miklos1) [inline docs/source/download.rst]: Done. Please have a look at documentation changes.

**Comment by miklos1** [comment]
@dham: Can you review the doc changes?

**Comment by dham** [comment]
Fine. Go for it.

**Pull request merged**